extensions/vertx-http/deployment/src/main/java/io/quarkus/vertx/http/deployment/webjar/WebJarUtil.java

```diff
@@ -0,0 +1,285 @@
+
+    private static final String CUSTOM_MEDIA_FOLDER = "META-INF/branding/";
+    private static final List<String> OVERRIDABLE_RESOURCES = Arrays.asList("logo.png", "favicon.ico", "style.css");
+    private static final String CSS = ".css";
+
+    private WebJarUtil() {
+    }
+
+    static Path copyResourcesForDevOrTest(CurateOutcomeBuildItem curateOutcomeBuildItem, ApplicationConfig config,
+            WebJarBuildItem webJar,
+            ResolvedDependency resourcesArtifact,
+            Path deploymentBasePath)
+            throws IOException {
+
+        Path deploymentPath = Files.createDirectories(deploymentBasePath);
+
+        PathTargetVisitor visitor = new PathTargetVisitor(deploymentPath);
+        copyResources(curateOutcomeBuildItem, config, webJar, resourcesArtifact, visitor);
+
+        return deploymentPath;
+    }
+
+    static Map<String, byte[]> copyResourcesForProduction(CurateOutcomeBuildItem curateOutcomeBuildItem,
+            ApplicationConfig config, WebJarBuildItem webJar,
+            ResolvedDependency resourcesArtifact)
+            throws IOException {
+
+        InMemoryTargetVisitor visitor = new InMemoryTargetVisitor();
+        copyResources(curateOutcomeBuildItem, config, webJar, resourcesArtifact, visitor);
+
+        return visitor.getContent();
+    }
+
+    private static void copyResources(CurateOutcomeBuildItem curateOutcomeBuildItem, ApplicationConfig config,
+            WebJarBuildItem webJar,
+            ResolvedDependency resourcesArtifact, WebJarResourcesTargetVisitor visitor)
+            throws IOException {
+        final ResolvedDependency userApplication = curateOutcomeBuildItem.getApplicationModel().getAppArtifact();
+
+        ClassLoader classLoader = WebJarUtil.class.getClassLoader();
+
+        resourcesArtifact.getContentTree().accept(webJar.getRoot(), new Consumer<PathVisit>() {
+            @Override
+            public void accept(PathVisit pathVisit) {
+                Path rootFolderToCopy = pathVisit.getPath();
+                if (!Files.isDirectory(rootFolderToCopy)) {
+                    return;
+                }
+
+                try {
+                    Files.walkFileTree(rootFolderToCopy, new ResourcesFileVisitor(visitor, rootFolderToCopy, resourcesArtifact,
+                            userApplication, config, classLoader, webJar));
+                } catch (IOException e) {
+                    throw new UncheckedIOException(e);
+                }
+            }
+        });
+    }
+
+    static ResolvedDependency getAppArtifact(CurateOutcomeBuildItem curateOutcomeBuildItem, ArtifactKey artifactKey) {
+        for (ResolvedDependency dep : curateOutcomeBuildItem.getApplicationModel().getDependencies()) {
+            if (dep.getArtifactId().equals(artifactKey.getArtifactId())
+                    && dep.getGroupId().equals(artifactKey.getGroupId())
+                    && dep.getClassifier().equals(artifactKey.getClassifier())) {
```

Comment:
That or `dep.getKey().equals(artifactKey)`

Reply:
I thought about it, too, but I don't see how `.equals` can return true.
`ResolvedDependency.getKey()` returns an instance of GACTV.

We only pass a GACT in here, since the version of the webjar is not static.
E.g. io.quarkus: quarkus-vertx-http-deployment is one of it.

Reply:
> I thought about it, too, but I don't see how `.equals` can return true. `ResolvedDependency.getKey()` returns an instance of GACTV.

No, it returns an `ArtifactKey`, which is `GACT`.

Reply:
yeah, you are obviously right. I guess I got confused, because ResolvedArtifactDependency extends GACTV. :D